test/functional/events_test.js

```diff
@@ -0,0 +1,53 @@
+var sinon = require('sinon');
+var should = require('should');
+
+var requestor = null;
+var smartsheet = null;
+
+describe('Client Unit Tests', function() {
+  beforeEach(function() {
+    requestor = require('../../lib/utils/httpRequestor.js').create({});
+    sinon.spy(requestor, 'get');
+    var client = require('../..');
+    smartsheet = client.createClient({accessToken:'1234', requestor: requestor});
+  });
+
+  afterEach(function() {
+    smartsheet = null;
+    requestor = null;
+  });
+
+  describe('#Events', function() {
+    it('should iterate through stream data until all data has been returned', function() {
+        // call events API with since query parameter
+        let moreEventsAvailable = true;
+        let options = {
+            queryParameters: {
+                since: '2018-01-09T17:41:05Z',
+                maxCount: 1
+            }
+        }
+
+        async function getEvents(moreEventsAvailable) {
+        do {
+            let result = await smartsheet.events.getEvents(options);
+            should(result.data).have.property('nextStreamPosition');
```

Comment:
This test isn't failing when expected. For example, the test doesn't fail even if you injected the lines of code below. The test doesn't make sense to me unless it fails and succeeds when expected. I suspect (please verify) the weird behavior is due to the fact that the test function is defined as asynchronous (async), and there is no other test in the entire SDK defined as asynchronous. I suggest you eliminate the async keyword and make it synchronous as the others.
            should(result.data).have.property('XXXXXX');
            should(YYYYYYYY).have.property('data');

Also, when debugging the test, I cannot succeed to make the Events API call work even when I give a valid Access Token. I am using the following way to define the smartsheet client:
smartsheet = client.createClient({accessToken:'MyToken', logLevel:'info', baseUrl:'https://api.plisboa.smart.ninja/2.0/'});

Please validate the test actually works when you give a valid Token and baseUrl but please show me the code that actually makes it work.